### PR TITLE
Linux tar and defconfig changes

### PR DESCRIPTION
**Pull request by osresearch**
This addresses issue #195 and #197 with some changes to how the Linux kernels are built.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -49,6 +49,29 @@ u-bmc:
 	go get
 	go build
 
+# Linux tree comes from the OpenBMC branch, not the official kernel.org
+LINUX_VERSION	:= 5.2.14
+LINUX_DIR	:= linux-$(LINUX_VERSION)
+LINUX_TAR	:= linux-$(LINUX_VERSION).tar.gz
+LINUX_URL	:= https://github.com/openbmc/linux/archive/v$(LINUX_VERSION).tar.gz
```
Comment (bluecmd) [inline Makefile]: I would assume the patches are applied by rebasing the patches upon branches. If that's the case the tags will always be prestine upstream kernel trees without any OpenBMC patches.

I spot-checked one tag and it appears to be the case. If you find counter-evidence I'm happy to consider this approach, but I think we will need an alternative.

File: platform/ast2400/linux.config.boot [hunk 2]
```diff
@@ -1,1602 +1,116 @@
...
-CONFIG_IRQ_WORK=y
-CONFIG_BUILDTIME_EXTABLE_SORT=y
-
-#
-# General setup
-#
-CONFIG_BROKEN_ON_SMP=y
-CONFIG_INIT_ENV_ARG_LIMIT=32
-# CONFIG_COMPILE_TEST is not set
 CONFIG_LOCALVERSION="-ubmc-boot"
```
Comment (bluecmd) [inline platform/ast2400/linux.config.boot]: Neat!

File: linux-patches/README.md [hunk 3]
```diff
@@ -1,7 +1,8 @@
 # Linux patches
 
 These are patches that are in the process of being upstreamed but needs to
-be patched now to support some critical functionallity.
+be patched now to support some critical functionallity. They will be
+applied automatically when the Linux kernel tar file is extracted.
```
Comment (bluecmd) [inline linux-patches/README.md]: It's not like they were documented before, but if you want feel free to mention menuconfig / saveconfig flow here.

File: Makefile (hunk 1 above)
Comment (osresearch) [reply Makefile]: I've also confirmed that the openbmc "release" 5.2.14 kernel is identical to the kernel.org 5.2.14. Wonder why they have their own tree of them...

So we need to see if there is an actual release tar file somewhere.

File: linux-patches/README.md (hunk 3 above)
Comment (osresearch) [reply linux-patches/README.md]: The top level `README.md` could use an overhaul to make it clearer how to build and configure for different platforms.  Updating the `$PLATFORM` variable in the makefile is the same hack we're doing in Heads and I wish there were a better way (like Kconfig, if it didn't have so much overhead to integrate)

**Pull request merged**